x/auth/account.go

```diff
@@ -121,12 +137,260 @@ func (acc *BaseAccount) SetSequence(seq int64) error {
+	return vestedCoins
+}
+
+// GetVestingCoins returns the total number of vesting coins. If no coins are
+// vesting, nil is returned.
+func (cva ContinuousVestingAccount) GetVestingCoins(blockTime time.Time) sdk.Coins {
+	return cva.OriginalVesting.Minus(cva.GetVestedCoins(blockTime))
+}
+
+// SpendableCoins returns the total number of spendable coins per denom for a
+// continuous vesting account.
+func (cva ContinuousVestingAccount) SpendableCoins(blockTime time.Time) sdk.Coins {
+	return cva.spendableCoins(cva.GetVestingCoins(blockTime))
+}
+
+// TrackDelegation tracks a desired delegation amount by setting the appropriate
+// values for the amount of delegated vesting, delegated free, and reducing the
+// overall amount of base coins.
+func (cva *ContinuousVestingAccount) TrackDelegation(blockTime time.Time, amount sdk.Coins) {
+	cva.trackDelegation(cva.GetVestingCoins(blockTime), amount)
+}
+
+//-----------------------------------------------------------------------------
+// Delayed Vesting Account
+
+var _ VestingAccount = (*DelayedVestingAccount)(nil)
+
+// DelayedVestingAccount implements the VestingAccount interface. It vests all
+// coins after a specific time, but non prior. In other words, it keeps them
+// locked until a specified time.
+type DelayedVestingAccount struct {
+	*BaseVestingAccount
+}
+
+func NewDelayedVestingAccount(
+	addr sdk.AccAddress, origCoins sdk.Coins, EndTime time.Time,
+) *DelayedVestingAccount {
+
+	baseAcc := &BaseAccount{
+		Address: addr,
+		Coins:   origCoins,
+	}
+
+	baseVestingAcc := &BaseVestingAccount{
+		BaseAccount:     baseAcc,
+		OriginalVesting: origCoins,
+		EndTime:         EndTime,
+	}
+
+	return &DelayedVestingAccount{baseVestingAcc}
+}
+
+// GetVestedCoins returns the total amount of vested coins for a delayed vesting
+// account. All coins are only vested once the schedule has elapsed.
+func (dva DelayedVestingAccount) GetVestedCoins(blockTime time.Time) sdk.Coins {
+	if blockTime.Unix() >= dva.EndTime.Unix() {
+		return dva.OriginalVesting
+	}
+
+	return nil
+}
+
+// GetVestingCoins returns the total number of vesting coins for a delayed
+// vesting account.
+func (dva DelayedVestingAccount) GetVestingCoins(blockTime time.Time) sdk.Coins {
+	return dva.OriginalVesting.Minus(dva.GetVestedCoins(blockTime))
+}
+
+// SpendableCoins returns the total number of spendable coins for a delayed
+// vesting account.
+func (dva DelayedVestingAccount) SpendableCoins(blockTime time.Time) sdk.Coins {
+	return dva.spendableCoins(dva.GetVestingCoins(blockTime))
+}
+
+// TrackDelegation tracks a desired delegation amount by setting the appropriate
+// values for the amount of delegated vesting, delegated free, and reducing the
+// overall amount of base coins.
+func (dva *DelayedVestingAccount) TrackDelegation(blockTime time.Time, amount sdk.Coins) {
+	dva.trackDelegation(dva.GetVestingCoins(blockTime), amount)
+}
+
+//-----------------------------------------------------------------------------
+// Codec
```

Comment:
shouldn't we split this file into the different account types + other (eg: `codec`)

Reply:
Perhaps. The definitions are short, but I'm indifferent overall 👍

Reply:
Actually, I think this might be overkill as the definitions are pretty succinct.